BrowserKit/Tests/ReduxTests/Utilities/FakeReduxMiddleware.swift

```diff
@@ -5,26 +5,33 @@
 import Foundation
 
 @testable import Redux
+
 class FakeReduxMiddleware {
     lazy var fakeProvider: Middleware<FakeReduxState> = { state, action in
-        switch action {
-        case FakeReduxAction.requestInitialValue:
+        switch action.actionType {
+        case FakeReduxActionType.requestInitialValue:
             let initialValue = self.generateInitialValue()
-            DispatchQueue.main.async {
-                store.dispatch(FakeReduxAction.initialValueLoaded(initialValue))
-            }
-        case FakeReduxAction.increaseCounter:
+            let action = FakeReduxAction(counterValue: initialValue,
+                                         windowUUID: windowUUID,
+                                         actionType: FakeReduxActionType.initialValueLoaded)
+            store.dispatch(action)
```

Comment:
This got me thinking, one of my big problems with readability is going from Action to middleware methods. (Right now there can be some logic done in the switchStatement as well as in a separate method. I was thinking it might be cleaner if the switch statement only had a call to a separate method with the context?

Current Example in the tab tray:
 ```
        case TabPanelAction.tabPanelDidLoad(let context):
            let isPrivate = context.boolValue
            let tabState = self.getTabsDisplayModel(for: isPrivate, shouldScrollToTab: true, uuid: uuid)
            let context = TabDisplayModelContext(tabDisplayModel: tabState, windowUUID: uuid)
            store.dispatch(TabPanelAction.didLoadTabPanel(context))

        case TabTrayAction.changePanel(let context):
            self.changePanel(context.panelType, uuid: uuid)

        case TabPanelAction.addNewTab(let context):
            let urlRequest = context.urlRequest
            let isPrivateMode = context.isPrivate
            self.addNewTab(with: urlRequest, isPrivate: isPrivateMode, for: uuid)

        case TabPanelAction.moveTab(let context):
            let originIndex = context.originIndex
            let destinationIndex = context.destinationIndex
            let isPrivate = context.isPrivate
            self.moveTab(state: state, from: originIndex, to: destinationIndex, isPrivate: isPrivate, uuid: uuid)
```

Proposed alternative:
```
        case TabPanelAction.tabPanelDidLoad(let context):
            self.tabPanelDidLoad(context)

        case TabTrayAction.changePanel(let context):
            self.changePanel(context)

        case TabPanelAction.addNewTab(let context):
            self.addNewTab(context)

        case TabPanelAction.moveTab(let context):
            self.moveTab(context)
```

Reply:
Just a suggestion, I can add it to your doc if you agree.